src/bidiMapper/modules/context/NavigationTracker.ts

```diff
@@ -400,64 +411,48 @@ export class NavigationTracker {
     }
 
     navigation.isFragmentNavigation = loaderId === undefined;
+  }
 
-    if (loaderId === undefined || this.#currentNavigation === navigation) {
-      // If the command's navigation is same-document or is already the current one,
-      // nothing to do.
-      return;
-    }
-
-    this.#currentNavigation.fail(
-      'navigation canceled by concurrent navigation',
-    );
-
-    navigation.start();
-    this.#currentNavigation = navigation;
+  frameStartedNavigating(
+    url: string,
+    loaderId: string,
+    navigationType: string,
+  ) {
+    this.#logger?.(LogType.debug, `frameStartedNavigating ${url}, ${loaderId}`);
 
-    if (this.#pendingNavigation === navigation) {
+    if (![loaderId, undefined].includes(this.#pendingNavigation?.loaderId)) {
```

Comment:
Can we not use this pattern but instead have two checks? 
Also with the inversion at the start it makes it even harder to understand. 
It looks confusing and you need to create an Array object, which is extra overhead.

Reply:
done